### DISCUSSION TITLE
Working with Rollup for UI library

### DISCUSSION THREAD

**Discussion by alexradulescu** [mayank99/ecsstatic#21] [Q&A]
Hi,

First of all awesome css-in-js library. 
If we would have had this from day 1 in react, I think most other css-in-js libraries would not have existed ;-) 
I love it.

Any thoughts how this would work with Rollup and UI libraries?
The use case I am thinking is for UI libraries, which many modern ones are using Rollup, including the ones I am using.
Besides my personal use, I saw libraries are moving more towards build-time css-in-js but also some like mantine.dev moving back to css modules to cancel out all the other issues coming with old css-in-js.

**Comment by mayank99** [comment]
Since Rollup plugins have the same API, I think it should work fine with Rollup? There are definitely vite-specific parts though, so you'll have to test it out yourself.

That being said, here's my $0.02:
- ~~Rollup is for bundling applications, not libraries.~~ EDIT: I've recently become aware of Rollup's history, and learned that this isn't true 😅
- Vite is basically Rollup with batteries included, so you might as well use Vite.
